### PR TITLE
feat(workflow): Add inbox details to Group serializer

### PR DESCRIPTION
**Pull request by iProgramStuff**
This PR adds the inbox information to the group serializer using the new `expand` idea.

The response will have a new key, `inbox` if the org has the inbox feature and the request includes `inbox` in the list of strings passed through the `expand` parameter (i.e. `expand="inbox"`). If the group is not in the inbox, this new key value will be `None`, and if it is in the inbox, it will be a dict containing `reason`, `reason_details`, and `date_added`.

### CODE REVIEW

File: src/sentry/api/serializers/models/group.py [hunk 1]
```diff
@@ -909,6 +912,20 @@ def _get_seen_stats(self, item_list, user):
                 )
         return time_range_result
 
+    def _get_inbox_stats(self, item_list):
+        group_ids = [g.id for g in item_list]
+        group_inboxes = GroupInbox.objects.filter(group__in=group_ids)
+        inbox_stats = {
+            gi.group.id: {
```
Comment (wedamija) [inline src/sentry/api/serializers/models/group.py:919]: `gi.group_id` saves you fetching it from the group.

File: src/sentry/api/serializers/models/group.py [hunk 2]
```diff
@@ -932,6 +949,14 @@ def get_attrs(self, item_list, user):
             for item in item_list:
                 if self.has_dynamic_issue_counts and self.conditions:
                     attrs[item].update({"filtered_stats": filtered_stats[item.id]})
+                if self.has_inbox:
+                    # Add inbox dict - either False, or label + reason, etc. "comments", "integration"
+                    inbox_stats = self._get_inbox_stats(item_list)
```
Comment (wedamija) [inline src/sentry/api/serializers/models/group.py:954]: You probably want this call to be outside of the for loop

File: src/sentry/api/serializers/models/group.py [hunk 3]
```diff
@@ -932,6 +949,14 @@ def get_attrs(self, item_list, user):
...
+                    if item.id in inbox_stats:
+                        attrs[item].update({"inbox": inbox_stats[item.id]})
+                    else:
+                        attrs[item].update({"inbox": None})
+
```
Comment (wedamija) [inline src/sentry/api/serializers/models/group.py:959]: `attrs[item].update({"inbox": inbox_stats.get(item.id)})

**Review by wedamija — COMMENTED** [review]
I'd like to start using the concept we discussed in backend TSC of passing the `expand` list of strings to decide which new values to include. That way we don't start including this in every response that serializes a group.

You basically already have this. I'd say that for this first pass, we'd rename `has_inbox` to `include_inbox` or something along those lines, and decide whether to fetch the inbox related things based off of that.

Then when you pass a value to `include_inbox`, it'd be something like `features.has("organizations:inbox", organization, actor=request.user) and 'group_inbox' in request.get('expand', [])`.

We'll probably generalize this more as we add it to more serializers, and probably end up passing the `expand` param to `serialize` instead, but this would be a good first pass

File: src/sentry/api/serializers/models/group.py (hunk 1 above)
Comment (iProgramStuff) [reply src/sentry/api/serializers/models/group.py:919]: Nice, thanks

File: src/sentry/api/serializers/models/group.py (hunk 2 above)
Comment (iProgramStuff) [reply src/sentry/api/serializers/models/group.py:954]: You know me so well.

File: src/sentry/api/endpoints/organization_group_index.py [hunk 4]
```diff
@@ -112,6 +112,10 @@ def get(self, request, organization):
             "organizations:dynamic-issue-counts", organization, actor=request.user
         )
 
+        include_inbox = features.has(
+            "organizations:inbox", organization, actor=request.user
+        ) and "group_inbox" in request.GET.get("expand", [])
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_group_index.py:117]: Should this be `request.GET.getlist()` Using .get() could get you a string if there is only one expand defined. Querystring parameters can be defined like `expand=one&expand=two` and `getlist()` will work consistently while `get()` will not from what I remember.

File: src/sentry/api/endpoints/organization_group_index.py [hunk 5]
```diff
@@ -112,6 +112,10 @@ def get(self, request, organization):
             "organizations:dynamic-issue-counts", organization, actor=request.user
         )
 
+        include_inbox = features.has(
+            "organizations:inbox", organization, actor=request.user
+        ) and "group_inbox" in request.GET.getlist("expand", [])
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_group_index.py:117]:
```suggestion
        ) and "inbox" in request.GET.getlist("expand", [])
```

Using `inbox` makes the request parameter and the attribute in the response match.

File: tests/snuba/api/endpoints/test_organization_group_index.py [hunk 6]
```diff
@@ -698,6 +701,91 @@ def test_skipped_fields(self):
...
+            add_group_to_inbox(event.group, GroupInboxReason.NEW)
+
+            query = u"server:example.com"
+            query += u" status:unresolved"
+            query += u" active_at:" + iso_format(before_now(seconds=350))
+            query += u" first_seen:" + iso_format(before_now(seconds=500))
+
+            self.login_as(user=self.user)
+            response = self.get_response(
+                sort_by="date", limit=10, query=query, expand=["group_inbox"]
```
Comment (markstory) [inline tests/snuba/api/endpoints/test_organization_group_index.py:727]:
```suggestion
                sort_by="date", limit=10, query=query, expand=["inbox"]
```

File: tests/snuba/api/endpoints/test_organization_group_index.py [hunk 7]
```diff
@@ -698,6 +701,91 @@ def test_skipped_fields(self):
...
+            snooze_details = {
+                "until": None,
+                "count": 3,
+                "window": None,
+                "user_count": None,
+                "user_window": 5,
+            }
+            add_group_to_inbox(event.group, GroupInboxReason.UNIGNORED, snooze_details)
+            response = self.get_response(
+                sort_by="date", limit=10, query=query, expand=["group_inbox"]
```
Comment (markstory) [inline tests/snuba/api/endpoints/test_organization_group_index.py:746]:
```suggestion
                sort_by="date", limit=10, query=query, expand=["inbox"]
```

File: tests/snuba/api/endpoints/test_organization_group_index.py [hunk 8]
```diff
@@ -698,6 +701,91 @@ def test_skipped_fields(self):
...
+            add_group_to_inbox(event.group, GroupInboxReason.NEW)
+
+            query = u"server:example.com"
+            query += u" status:unresolved"
+            query += u" active_at:" + iso_format(before_now(seconds=350))
+            query += u" first_seen:" + iso_format(before_now(seconds=500))
+
+            self.login_as(user=self.user)
+            response = self.get_response(
+                sort_by="date", limit=10, query=query, expand="group_inbox"
```
Comment (markstory) [inline tests/snuba/api/endpoints/test_organization_group_index.py:779]:
```suggestion
                sort_by="date", limit=10, query=query, expand="inbox"
```

**Review by markstory — APPROVED** [review]
Looks good to me. I left a few suggestions on a small consistency bit but I don't think they are blockers.

**Pull request merged**